### PR TITLE
implementation of 3.1 support deserialization and validation of non JsonSchema fields

### PR DESCRIPTION
**Pull request by gracekarina**
implementation of 3.1 support deserialization and validation of non JsonSchema fields

### CODE REVIEW

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java [hunk 1]
```diff
@@ -369,12 +375,25 @@ public OpenAPI parseRoot(JsonNode node, ParseResult result, String path) {
 				openAPI.setExtensions(extensions);
 			}
 
+			if (result.isOpenapi31()) {
+				value = getString("jsonSchemaDialect", rootNode, false, location, result);
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:379]: it makes sense to check if the string is a URL, according to spec

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java [hunk 2]
```diff
@@ -386,6 +405,15 @@ public OpenAPI parseRoot(JsonNode node, ParseResult result, String path) {
 		return openAPI;
 	}
 
+	private void validateReservedKeywords(Map<String, Set<String>> specKeys, String key, String location, ParseResult result) {
+		if(result.isOpenapi31() && specKeys.get("RESERVED_KEYWORDS").stream()
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:409]: better to add a new boolean flag in result, like `oaiInternal` or `oaiAuthor` to allow for any key prefix if the author is indeed OAI

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java [hunk 3]
```diff
@@ -2153,9 +2195,9 @@ public SecurityScheme getSecurityScheme(ObjectNode node, String location, ParseR
 		}
 
 		boolean descriptionRequired, bearerFormatRequired, nameRequired, inRequired, schemeRequired, flowsRequired,
-				openIdConnectRequired;
+				openIdConnectRequired, mutualTLSRequired;
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:2198]: mutualTLSRequired not used, what is this about?

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java [hunk 4]
```diff
@@ -2397,6 +2442,7 @@ public Discriminator getDiscriminator(ObjectNode node, String location, ParseRes
 					if (extensions != null && extensions.size() > 0) {
 						discriminator.setExtensions(extensions);
 					}
+					//validateReservedKeywords(keys,key, location, result);
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:2445]: why is this commented?

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java [hunk 5]
```diff
@@ -4012,6 +4069,11 @@ public boolean isOpenapi31() {
 				String message = "attribute " + location + l.key + " is repeated";
 				messages.add(message);
 			}
+			for (Location l : reserved) {
+				String location = l.location.equals("") ? "" : l.location + ".";
+				String message = "attribute " + location + l.key + " is reserved by The OpenAPI Iniciative";
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:4074]: Typo: `Iniciative` -> `Initiative`

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java [hunk 6]
```diff
@@ -2821,6 +2867,7 @@ at the moment path passed as string (basePath) from upper components can be both
 			if (!specKeys.get("SCHEMA_KEYS").contains(key) && !key.startsWith("x-")) {
 				result.extra(location, key, node.get(key));
 			}
+			validateReservedKeywords(specKeys, key, location, result);
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:2870]: This is handling 3.0, so not applicable. Also I believe this restriction is not applied to Json Schema also in 3.1

**Review by frantuma — CHANGES_REQUESTED** [review]
On top of comments, please add a test demonstrating that a spec with all new constructs parses ok, with the fields populated

File: modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java (hunk 6 above)
Comment (gracekarina) [reply modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java:2870]: I'll remove the validation from there but I think in the method we'll handle 3.1 schema(JsonSchema) we may find also extensions with `x-` or without.  
`This object MAY be extended with Specification Extensions, though as noted, additional properties MAY omit the x- prefix within this object.`

**Comment by frantuma** [comment]
@gracekarina if I get this correctly, this PR is replaced by #1617 and should therefore be closed?

**Pull request closed without merging**